app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java

```diff
@@ -67,18 +105,31 @@
     private int leastCount, seekMax, seekMin;
     private String unit;
 
+    public enum WaveConst {WAVETYPE, WAVE1, WAVE2, SQ1, SQ2, SQ3, SQ4, FREQUENCY, PHASE, DUTY}
+
+    public enum WaveData {
+        FREQ_MIN(10), DUTY_MIN(0), PHASE_MIN(0), FREQ_MAX(5000), PHASE_MAX(360), DUTY_MAX(100);
+
+        public final int value;
+
+        WaveData(final int v) {
+            value = v;
+        }
+
+        public final int getValue() {
+            return value;
+        }
+    }
+
     //const values
     public static final int SIN = 1;
     public static final int TRIANGULAR = 2;
     public static final int SQUARE = 3;
-    public static final String WAVETYPE = "wavetype";
-    public static final String WAVE1 = "wave1";
-    public static final String WAVE2 = "wave2";
-    public static final String FREQUENCY = "frequency";
-    public static final String PHASE = "phase";
-    public static final String DUTY = "duty";
-    private static String btn_Active = "";
-    private static String prop_active = "";
+
+    private WaveConst waveBtnActive, pwmBtnActive, prop_active;
+    private static boolean waveMonSelected;
+    private TextView activePropTv = null;
+    private TextView activePwmPinTv = null;
```

Comment:
Where is this view bidden to? Is it in working progress?

Reply:
That just textview holder which assumes the textview already on the screen so as to changes their values.

Reply:
Yes it is in working state.